### PR TITLE
Fix for previewing on not current window

### PR DESCRIPTION
**Pull request by aki2o**
`consult--jump-preview` could not work with the feature that manages `display-buffer-function` like [e2wm.el](https://github.com/kiwanami/emacs-window-manager).
That's because `consult--buffer-action` don't have a consideration for the case that `current-buffer` won't be switched.

So, I tried to fix this trouble and it looks to work well with this fix.

### CODE REVIEW

File: consult.el [hunk 1]
```diff
@@ -469,7 +469,7 @@ called from the corresponding command. Note that the options depend on
 the private `consult--read' API and should not be considered as stable
 as the public API.")
 
-(defvar consult--buffer-display #'switch-to-buffer
+(defvar consult--buffer-display #'display-buffer
```
Comment (aki2o) [inline consult.el:472]: I'm using e2wm.el and it must support `switch-to-buffer`.
However, it happens to collapse window layout in my environment.

Anyway, I believe `display-buffer` is more appropriate for this package role. So, I hope this fix be included

File: consult.el [hunk 2]
```diff
@@ -1410,10 +1423,10 @@ The function can be used as the `:state' argument of `consult--read'."
...
                 (narrow-to-region saved-min saved-max)
                 (goto-char saved-pos)))
-            ;; Handle positions with overlay information
-            (consult--jump-1 (or (car-safe cand) cand))
-            (setq invisible (consult--invisible-open-temporarily)
-                  overlays
+          ;; Handle positions with overlay information
+          (consult--with-jump (or (car-safe cand) cand)
+            (setq consult--preview-previous-invisibles (consult--invisible-open-temporarily)
+                  consult--preview-previous-overlays
```
Comment (aki2o) [inline consult.el:1429]: I feel `invisible` should be defined for not local but global because `void variable` error happened.

File: consult.el [hunk 3]
```diff
@@ -1357,22 +1357,36 @@ See `isearch-open-necessary-overlays' and `isearch-open-overlay-temporary'."
...
+           (with-selected-window w (funcall f))
+         (funcall f)))))
+
+(defvar consult--preview-previous-invisibles nil)
+(defvar consult--preview-previous-overlays nil)
 
 (defun consult--jump (pos)
   "Push current position to mark ring, go to POS and recenter."
+  (mapc #'funcall consult--preview-previous-invisibles)
+  (mapc #'delete-overlay consult--preview-previous-overlays)
```
Comment (aki2o) [inline consult.el:1389]: it looks like the variables need to be cleared here as well.
And I guess there is a timing that's more appropriate to do this procedure but I couldn't find it.

File: consult.el [hunk 4]
```diff
@@ -1357,22 +1357,36 @@ See `isearch-open-necessary-overlays' and `isearch-open-overlay-temporary'."
...
+                     (t
+                      (get-buffer-window (current-buffer)))))
+            (f (lambda ()
+                 (unless (= (goto-char ,pos) (point))
+                   (widen) ;; Widen if we cannot jump to the position (idea from flycheck-jump-to-error)
+                   (goto-char ,pos))
+                 ,@body
+                 (run-hooks 'consult-after-jump-hook))))
+       (if (window-live-p w)
+           (with-selected-window w (funcall f))
```
Comment (aki2o) [inline consult.el:1380]: Here is taking care of the case like

* `current-buffer` is not the target buffer
* the window of the target buffer is not selected window

File: consult.el (hunk 1 above)
Comment (minad) [reply consult.el:472]: No, switch-to-buffer is the better choice here. You can set switch-to-buffer-obey-display-actions or change the variable in your configuration. Please change back.

File: consult.el (hunk 2 above)
Comment (minad) [reply consult.el:1429]: Yes, this is wrong. Please revert the change to the overlays variable.

File: consult.el [hunk 5]
```diff
@@ -1357,22 +1357,36 @@ See `isearch-open-necessary-overlays' and `isearch-open-overlay-temporary'."
...
+         (funcall f)))))
+
+(defvar consult--preview-previous-invisibles nil)
+(defvar consult--preview-previous-overlays nil)
```
Comment (minad) [inline consult.el:1384]: Remove these variables.

**Comment by minad** [comment]
I close this PR in favor of #671. We should first fix consult--buffer-preview to handle display-buffer properly before trying to fix consult--jump-preview.

**Pull request closed without merging**